### DISCUSSION TITLE
librewolf: command not found (AppImage)

### DISCUSSION THREAD

**Discussion by NoRainfallDuckDog** [netblue30/firejail#4500] [Q&A] [answered]
I didn't use the flatpak method but used the AppImage.
It works, but when issuing `firejail librewolf`, it won't launch.
Getting the the following errors:

The following errors remain:

```
Reading profile /etc/firejail/default.profile
Reading profile /etc/firejail/disable.common.inc
Reading profile /etc/firejail/disable.passwdmgr.inc
Reading profile /etc/firejail/disable.programs.inc

Warning: networking feature is disabled in Firejail configuration file

** Note: you can use --noprofile to disable default.profile **

/bin/bash: librewolf: command not found
```

There are necessary Firefox ".profile" files but there are none for LW under /etc/firejail.

It seems that it is looking for the profiles for Librewolf.

Is there a repository for the ".profile" files that I can just copy to the /etc/firejail directory for Librewolf? 

-----

Edit by @kmk3: Formatting/grammar.

**Comment by kmk3** [comment]
@nntn-nntn [on Sep 2](https://github.com/netblue30/firejail/discussions/4500#discussion-3551289):

> I didn't use the flatpak method but used the AppImage.  It works, but when
> issuing `firejail librewolf`, it won't launch.  Getting the the following
> errors:
> 
> The following errors remain:
> 
> ```
> Reading profile /etc/firejail/default.profile
> Reading profile /etc/firejail/disable.common.inc
> Reading profile /etc/firejail/disable.passwdmgr.inc
> Reading profile /etc/firejail/disable.programs.inc
> 
> Warning: networking feature is disabled in Firejail configuration file
> 
> ** Note: you can use --noprofile to disable default.profile **
> 
> /bin/bash: librewolf: command not found
> ```

If you're trying to run an AppImage file, you should specify its path.
Example:

```sh
firejail --appimage --profile=librewolf ./librewolf.AppImage
```

See firejail(1) for some examples.

> There are necessary Firefox ".profile" files but there are none for LW under
> /etc/firejail.
> 
> It seems that it is looking for the profiles for Librewolf.
> 
> Is there a repository for the ".profile" files that I can just copy to the
> /etc/firejail directory for Librewolf?

There is a librewolf.profile on the current firejail release (0.9.66):

<https://github.com/netblue30/firejail/blob/0.9.66/etc/profile-a-l/librewolf.profile>

What is your firejail version and distro?

**Comment by NoRainfallDuckDog** [comment]
I didn't think AppImage is just an image and not a program.

I am using Linux Mint 20.2 recent version
 
I issued the following command:
Firejail --version                        -------> It  shows firejail version 0.9.62
sudo apt install firejail                ------> It still shows 0.9.62

It didn't upgrade the version to 9.66
How can I upgrade the version? 

How do I download the profile? Is it copy and paste?
If it is copy and paste then what is the command to create a profile? Is it "sudo nano"?
I was able to use ---> sudo nano <filename> <----- to make change to one of the profiles from /etc
Once the profile"librewolf.profile" is created and put in "/etc/firejail", can I just issue the command "firejail librewolf"?

There are suggestions to issue the lengthy command "firejail" to include the appimage.

**Reply by kmk3** [reply, marked as answer]
@nntn-nntn [on Sep 7](https://github.com/netblue30/firejail/discussions/4500#discussioncomment-1292661):

> I didn't think AppImage is just an image and not a program.

It's both.

> I am using Linux Mint 20.2 recent version
> 
> I issued the following command:
> Firejail --version -------> It shows firejail version 0.9.62
> sudo apt install firejail ------> It still shows 0.9.62
> 
> It didn't upgrade the version to 9.66
> How can I upgrade the version?

From [README.md][1]:

> The firejail 0.9.52-LTS version is deprecated. On Ubuntu 18.04 LTS users are
> advised to use the
> [PPA](https://launchpad.net/~deki/+archive/ubuntu/firejail). On Debian buster
> we recommend to use the
> [backports](https://packages.debian.org/buster-backports/firejail) package.

@reinerh For users on >18.04 that want to use the latest stable, is the PPA the
solution?  Is this supported?

> How do I download the profile? Is it copy and paste?

Yes, it can be copied to /etc/firejail or ~/.config/firejail.

Though note that mixing and matching profiles and firejail versions may cause
issues (e.g.: a profile that uses commands only supported by a newer firejail
version).

> If it is copy and paste then what is the command to create a profile? Is it
> "sudo nano"?  I was able to use ---> sudo nano <----- to make change to one
> of the profiles from /etc

Yes, that should work.

> Once the profile"librewolf.profile" is created and put in "/etc/firejail",
> can I just issue the command "firejail librewolf"?

Only for normal programs (that exist in `$PATH`), not AppImages.  For AppImages
you should pass `--appimage` and specify the file path, like in the example in
my previous comment.

> There are suggestions to issue the lengthy command "firejail" to include the
> appimage.

I don't understand what you mean.  If you mean that the command is too lengthy,
you can put it in a shell script and execute that instead.  Example:

```console
$ cat ~/bin/firejail-librewolf
#!/bin/sh
exec firejail --appimage --profile=librewolf /path/to/librewolf.appimage
```

Then you can execute it as `~/bin/firejail-librewolf`, or just as
`firejail-librewolf` if ~/bin is in your `$PATH`.

[1]: https://github.com/netblue30/firejail/blob/3a333864ca67443664b17a3838bd53ed94c09714/README.md

**Reply by reinerh** [reply]
> From [README.md](https://github.com/netblue30/firejail/blob/3a333864ca67443664b17a3838bd53ed94c09714/README.md):
> 
> > The firejail 0.9.52-LTS version is deprecated. On Ubuntu 18.04 LTS users are
> > advised to use the
> > [PPA](https://launchpad.net/~deki/+archive/ubuntu/firejail). On Debian buster
> > we recommend to use the
> > [backports](https://packages.debian.org/buster-backports/firejail) package.
> 
> @reinerh For users on >18.04 that want to use the latest stable, is the PPA the
> solution? Is this supported?

Yes, the PPA provides the latest firejail stable for Ubuntu 18.04, 20.04, 21.04 and 21.10.
They are compiled by the build servers for these specific Ubuntu releases (because of that, they require minor changes to support older build tools (e.g. debhelper)). So yes, they are supported (though I don't know the differences between Ubuntu and Mint, but I would expect that it works on Mint as well).

**Reply by kmk3** [reply]
@reinerh [on Sep 8](https://github.com/netblue30/firejail/discussions/4500#discussioncomment-1298366):

> > @reinerh For users on >18.04 that want to use the latest stable, is the PPA
> > the solution? Is this supported?
> 
> Yes, the PPA provides the latest firejail stable for Ubuntu 18.04, 20.04,
> 21.04 and 21.10.  They are compiled by the build servers for these specific
> Ubuntu releases (because of that, they require minor changes to support older
> build tools (e.g. debhelper)). So yes, they are supported (though I don't
> know the differences between Ubuntu and Mint, but I would expect that it
> works on Mint as well).

I see; thanks for the details.

**Comment by NoRainfallDuckDog** [comment]
My proficiency in this area is so-so. 

It seems that to use the provided profile, I will have to upgrade it.

Per your suggestions, that mixing and matching profiles and firejail versions may cause
issues (e.g.: a profile that uses commands only supported by a newer firejail
version).

I will try the shell script. 

Thanks for the responses!
